src/components/utils/TimedMount.tsx

```diff
@@ -0,0 +1,55 @@
+import { Container, EffectProps, StateUpdater } from 'constate'
+import * as React from 'react'
+import { Mount, Unmount } from 'react-lifecycle-components'
+import { createTimer, Timer } from '../../utils/timer'
+
+interface State {
+  hasFired: boolean
+  timer?: Timer
+}
+
+interface Effects {
+  abort: () => StateUpdater<State>
+  initiateTimer: () => (props: EffectProps<State>) => void
+}
+
+interface Props {
+  duration: number
+  children: (props: { hasFired: boolean }) => React.ReactNode
+  onFire?: () => void
+}
+
+export const TimedMount: React.SFC<Props> = ({
```

Comment:
FYI this is very similar to delayed :) https://github.com/Wildhoney/ReactDelayed#readme

Reply:
I guess this naming is a bit off? Because technically it kind of does the same thing as `react-transition-group` right? (optionally gives you an `entered` state)
We should probably switch it out for `react-transition-group` right?

Reply:
It depends on what you want to achieve, `react-delayed` is a bit nicer if you want to delay complicated renders due to performance reasons, I usually end up in using a combination between `react-transition-group` and `delayed` 😄